core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -918,8 +913,16 @@ public Result visit(TableFunctionScan e) {
     String alias = SqlValidatorUtil.getAlias(node, -1);
     final String lowerName = name.toLowerCase(Locale.ROOT);
     if (lowerName.startsWith("expr$")) {
-      // Put it in ordinalMap
-      ordinalMap.put(lowerName, node);
+      final ArrayDeque<Frame> clone = new ArrayDeque<>(stack);
+      clone.pop();
+      final Frame parent = clone.peek();
+      if (parent != null && parent.r instanceof Union) {
+        // For case: Project(Union(expression without alias))
```

Comment:
Not only `Union`, i believe all the `SetOp` and `BiRel` can have such a simplification.

Reply:
Hi @danny0405 , I have updated the solution and fix the code conflict.